### PR TITLE
Add CNI reconciler for docker

### PR DESCRIPTION
**Pull request by vivek-koppuru**
*Issue #, if available:*

*Description of changes:*
Implement the provider reconciler for cilium networking for the docker provider

*Testing (if applicable):*
unit testing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by vivek-koppuru** [comment]
/hold

Still working on some refactoring of repeated code

File: internal/test/testdata.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
+package test
```
Comment (taneyland) [inline internal/test/testdata.go:1]: Super useful addition for our unit tests!

**Review by taneyland — APPROVED** [review]

**Pull request merged**